lib/charms/grafana_agent/v0/cos_machine.py

```diff
@@ -1,8 +1,166 @@
 #!/usr/bin/env python3
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
 
-"""Library for the cos_machine relation interface."""
+r"""## Overview.
+
+This document explains how to use the two principal objects this library provides:
+
+- `COSMachineProvider`: This object can be used by any Charmed Operator which needs to
+send telemetry, such metrics, logs and dashboards to Prometheus, Loki and Grafana through
+Grafana Agent machine charm.
+
+- `COSMachineConsumer`: This object can be used by any Charmed Operator that needs to implement
+the consumer side of this `cos_machine` interface, for instance this Grafana Agent machine
+Charmed Operator.
+
+
+## COSMachineProvider Library Usage
+
+Grafana Agent machine Charmed Operator interacts with its clients using the cos_machine library.
+Charms seeking to send telemetry, must do so using the `COSMachineProvider` object from
+this charm library.
+
+Using the `COSMachineProvider` object only requires instantiating it,
+typically in the `__init__` method of your charm (the one which sends telemetry).
+
+The constructor of `COSMachineProvider` has only one required and eight optional parameters:
+
+```python
+    def __init__(
+        self,
+        charm: CharmType,
+        relation_name: str = DEFAULT_RELATION_NAME,
+        metrics_endpoints: Optional[List[dict]] = None,
+        metrics_rules_dir: str = "./src/prometheus_alert_rules",
+        logs_rules_dir: str = "./src/loki_alert_rules",
+        recurse_rules_dirs: bool = False,
+        logs_slots: Optional[List[str]] = None,
```

Comment:
Slightly out of scope:
```suggestion
        log_slots: Optional[List[str]] = None,
```